### PR TITLE
Add 'dropdown with edit'.

### PR DESCRIPTION
**Pull request by ceball**
Before I go further, would be great to get feedback on whether or not this is usable*, and if the approach will work for parambokeh.

Note: can't use with ClassSelector yet (#49, https://github.com/ioam/param/issues/162).

* With your own imagined better UI (no work done on UI so far).

### CODE REVIEW

File: paramnb/widgets.py [hunk 1]
```diff
@@ -252,6 +258,51 @@ def get_state(self, key=None, drop_defaults=False):
...
+# Some things are routed to/from the Dropdown, other to/from the
+# composite HBox.
+class DropdownWithEdit(ipywidgets.Widget):
+    """
+    Dropdown, but displays an edit button if the current selection is
+    a parameterized object.
+    """
+
+    # I couldn't figure out which widget class actually declares the
+    # value trait. Might not be needed in in ipywidgets>6 because I
```
Comment (jbednar) [inline paramnb/widgets.py]: in in

**Comment by jbednar** [comment]
Looks great to me (apart from the HACK), and I'll be happy to see this finished up and merged.

**Comment by jbednar** [comment]
Are you merging the demo notebook into the main notebook now?  It would be good to demonstrate this functionality somewhere...

**Comment by ceball** [comment]
Sorry, meant to note in the commit but must've forgotten. I added a new notebook for 'additional features', and included the previously not-advertised ViewParameters notebook in it. 

Not great but it's better than nothing.

**Pull request merged**